### PR TITLE
Feature/watchlist

### PR DESCRIPTION
**Pull request by kosovych**

### CODE REVIEW

File: src/assets/styles/app.scss [hunk 1]
```diff
@@ -80,3 +80,10 @@ body,
 .d-none {
   display: none !important;
 }
+
```
Comment (VladWithCookies) [inline src/assets/styles/app.scss:83]: extra new line

File: src/containers/Header/container.js [hunk 2]
```diff
@@ -1,20 +1,61 @@
...
 
-import { logout } from 'Store/auth/actions'
+import { logout, getUser as getUserAction } from 'Store/auth/actions'
 import HeaderComponent from './component'
 
-const Header = ({ onLogout }) => (
-  <HeaderComponent onLogout={onLogout} />
-)
+class Header extends Component {
+  componentWillMount() {
```
Comment (VladWithCookies) [inline src/containers/Header/container.js:9]: `componentWillMount` is deprecated, please use `componentDidMount`

File: src/containers/WatchList/DeleteMovie/component.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { DeleteOutlined } from '@ant-design/icons'
+
+const DeleteMovie = ({ onClick }) => (
+  <button
+    onClick={onClick}
+    type="button"
+    className="reset-btn"
+  >
+    <DeleteOutlined
+      key="open-modal"
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/component.js:12]: do we really need `key` here?

File: src/containers/WatchList/DeleteMovie/container.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    type: 'error',
+    onClose() { cleanupRemoveMovieErrors(movieId) }
+  }
+  const error = removeMovieErrors.includes(movieId) && openNotification(config)
+  return (
+    <DeleteMovieComponent
+      onClick={() => Modal.confirm({
+        title: 'Do you want to delete movie from watchlist?',
+        onOk() { removeMovie(movieId, title) }
+      })}
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:33]: extract this to the method like `handleDeleteMovie`

File: src/store/auth/operations/getUserOperation.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import { ACCOUNT } from 'Constants'
+import { GET_USER } from '../types'
+import { storeUser } from '../actions'
+
+export const getUserOperation = createLogic({
+  type: GET_USER,
+  latest: true,
+  async process({ action, axios }, dispatch, done) {
+    const { sessionID } = action
+    const userReq = await axios.get(ACCOUNT, { params: { session_id: sessionID } })
```
Comment (VladWithCookies) [inline src/store/auth/operations/getUserOperation.js:13]: `userReq` -> `response`

File: src/store/watchList/operations/getWatchListOperation.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import { GET_MOVIES } from '../types'
+import { requestMoviesStart, requestMoviesSuccess, requestMoviesError } from '../actions'
+
+export const getWatchListOperation = createLogic({
+  type: GET_MOVIES,
+  latest: true,
+  async process({ action, axios, getState }, dispatch, done) {
+    const { params = {} } = action
+    params.session_id = sessionIdSelector(getState())
+    const url = `/account/${userIdSelector(getState())}/watchlist/movies`
```
Comment (VladWithCookies) [inline src/store/watchList/operations/getWatchListOperation.js:16]: `getState()` is used twice. lets extract it to the variable

File: src/store/watchList/reducer.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      return {
+        ...state,
+        entries: without(state.entries, action.movieId)
+      }
+    case SET_REMOVE_MOVIE_ERROR:
+      return {
+        ...state,
+        meta: {
+          ...state.meta,
+          removeErrors: concat(state.meta.removeErrors || [], [action.movieId])
```
Comment (VladWithCookies) [inline src/store/watchList/reducer.js:53]: ```state.meta.removeErrors?.push(action.movieId)```

File: src/assets/styles/app.scss (hunk 1 above)
Comment (kosovych) [reply src/assets/styles/app.scss:83]: ✅

File: src/containers/Header/container.js (hunk 2 above)
Comment (kosovych) [reply src/containers/Header/container.js:9]: ✅

File: src/containers/WatchList/DeleteMovie/component.js (hunk 3 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/component.js:12]: no

File: src/containers/WatchList/DeleteMovie/container.js (hunk 4 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:33]: ✅

File: src/store/auth/operations/getUserOperation.js (hunk 5 above)
Comment (kosovych) [reply src/store/auth/operations/getUserOperation.js:13]: ✅

File: src/store/watchList/operations/getWatchListOperation.js (hunk 6 above)
Comment (kosovych) [reply src/store/watchList/operations/getWatchListOperation.js:16]: ✅

File: src/store/watchList/reducer.js (hunk 7 above)
Comment (kosovych) [reply src/store/watchList/reducer.js:53]: @VladWithCookies because `removeErrors` can be undefined and in this case I need `[]`

Comment (VladWithCookies) [reply src/store/watchList/reducer.js:53]: try to use optional chaining like on my example above. i think this should work )

Comment (kosovych) [reply src/store/watchList/reducer.js:53]: @VladWithCookies https://codepen.io/y_kosovych/pen/gOLmEWL If I not mistaken It doesn’t help

File: src/containers/WatchList/DeleteMovie/component.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { DeleteOutlined } from '@ant-design/icons'
+
+const DeleteMovie = ({ onClick }) => (
+  <button
+    onClick={onClick}
+    type="button"
+    className="reset-btn"
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/component.js:9]: please add `aria-label` )

File: src/containers/WatchList/DeleteMovie/container.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+  removeMovie as removeMovieAction,
+  cleanupRemoveMovieErrors as cleanupRemoveMovieErrorsAction
+} from 'Store/watchList/actions'
+import DeleteMovieComponent from './component'
+
+class DeleteMovie extends Component {
+  onDeleteMovie = () => {
+    const { removeMovie, movieId, title } = this.props
+    return Modal.confirm({
+      title: 'Do you want to delete movie from watchlist?',
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:18]: Do you want to delete ***this*** movie from ***the*** watchlist

File: src/containers/WatchList/DeleteMovie/container.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+      onOk() { removeMovie(movieId, title) }
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:19]: `onOk = () => removeMovie(movieId, title)`

File: src/containers/WatchList/DeleteMovie/container.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+      movieId,
+      title,
+      removeMovieErrors,
+      cleanupRemoveMovieErrors
+    } = this.props
+    const config = {
+      message: 'Error',
+      description: `Cant remove "${title}" from Watchlist`,
+      type: 'error',
+      onClose() { cleanupRemoveMovieErrors(movieId) }
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:34]: arrow function

File: src/containers/WatchList/DeleteMovie/container.js [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    const error = removeMovieErrors.includes(movieId) && openNotification(config)
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:36]: getter

File: src/containers/WatchList/DeleteMovie/container.js [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+    return (
+      <DeleteMovieComponent
+        onClick={this.onDeleteMovie}
+      >
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:40]: one line

File: src/containers/WatchList/DeleteMovie/container.js [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+  render() {
+    const {
+      movieId,
+      title,
+      removeMovieErrors,
+      cleanupRemoveMovieErrors
+    } = this.props
+    const config = {
+      message: 'Error',
+      description: `Cant remove "${title}" from Watchlist`,
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:32]: Cant remove "${title}" from ***the*** Watchlist

File: src/containers/WatchList/DeleteMovie/container.js [hunk 15]
```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Modal } from 'antd'
+import { connect } from 'react-redux'
+import { openNotification } from 'Utils'
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:5]: this import is related to the our modules, not 3rd party

File: src/store/watchList/reducer.js (hunk 7 above)
Comment (VladWithCookies) [reply src/store/watchList/reducer.js:53]: ok, you're right. let's omit `[]` in 2nd argument in `concat` then.

File: src/containers/WatchList/DeleteMovie/container.js [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  render() {
+    const {
+      movieId,
+      title,
+      removeMovieErrors,
+      cleanupRemoveMovieErrors
+    } = this.props
+    const config = {
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:30]: getter

File: src/store/watchList/reducer.js (hunk 7 above)
Comment (kosovych) [reply src/store/watchList/reducer.js:53]: @VladWithCookies but it's something like default value, I can't omit it

File: src/components/MovieItem/component.js [hunk 17]
```diff
@@ -2,30 +2,47 @@ import React from 'react'
...
-    cover={(
-      <img
-        alt={title}
-        src={`https://image.tmdb.org/t/p/w500/${poster}`}
-        width="500"
-        height="750"
+const Movie = ({
+  poster, overview, title, action: Action, movieId
+}) => {
+  const actionsArr = Action ? ([<Action
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/components/MovieItem/component.js:8]: ✅

File: src/containers/WatchList/DeleteMovie/component.js (hunk 8 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/component.js:9]: ✅

File: src/containers/WatchList/DeleteMovie/container.js (hunk 9 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:18]: ✅

File: src/containers/WatchList/DeleteMovie/container.js (hunk 10 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:19]: 1. `onOk: () => removeMovie(movieId, title)` *
2. It doesn’t works this way

File: src/containers/WatchList/DeleteMovie/container.js (hunk 11 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:34]: ✅

File: src/containers/WatchList/DeleteMovie/container.js (hunk 13 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:40]: ✅

File: src/containers/WatchList/DeleteMovie/container.js (hunk 14 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:32]: ✅

File: src/containers/WatchList/DeleteMovie/container.js (hunk 15 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:5]: ✅

File: src/containers/WatchList/DeleteMovie/container.js (hunk 16 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:30]: ✅

File: src/containers/WatchList/DeleteMovie/container.js (hunk 12 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:33]: ✅

File: src/store/watchList/reducer.js (hunk 7 above)
Comment (VladWithCookies) [reply src/store/watchList/reducer.js:53]: i meaned this: `concat(state.meta.removeErrors || [], action.movieId)`

Comment (kosovych) [reply src/store/watchList/reducer.js]: ✅

File: src/containers/WatchList/DeleteMovie/container.js [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+import {
+  removeMovie as removeMovieAction,
+  cleanupRemoveMovieErrors as cleanupRemoveMovieErrorsAction
+} from 'Store/watchList/actions'
+import DeleteMovieComponent from './component'
+
+class DeleteMovie extends Component {
+  onDeleteMovie = () => {
+    const { removeMovie, movieId, title } = this.props
+    return Modal.confirm({
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:17]: do we need to return it?

File: src/containers/WatchList/DeleteMovie/container.js [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+    const {
+      movieId, title, cleanupRemoveMovieErrors, removeMovieErrors
+    } = this.props
+    const config = {
+      message: 'Error',
+      description: `Can't remove "${title}" the from Watchlist`,
+      type: 'error',
+      onClose: () => cleanupRemoveMovieErrors(movieId)
+    }
+    const error = removeMovieErrors.includes(movieId) && openNotification(config)
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:33]: it is looks like `openNotification` should be in the operation

File: src/containers/WatchList/DeleteMovie/container.js [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+    return error
+  }
+
+  render() {
+    return (
+      <DeleteMovieComponent onClick={this.onDeleteMovie}>
+        {this.error}
```
Comment (VladWithCookies) [inline src/containers/WatchList/DeleteMovie/container.js:40]: you don't need this line

Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:40]: @VladWithCookies 
image

File: src/containers/WatchList/DeleteMovie/container.js (hunk 19 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:33]: @VladWithCookies 
image

Comment (VladWithCookies) [reply src/containers/WatchList/DeleteMovie/container.js:33]: Ок

File: src/containers/WatchList/DeleteMovie/container.js (hunk 20 above)
Comment (VladWithCookies) [reply src/containers/WatchList/DeleteMovie/container.js:40]: Ок

File: src/containers/WatchList/DeleteMovie/container.js (hunk 18 above)
Comment (kosovych) [reply src/containers/WatchList/DeleteMovie/container.js:17]: no

**Review by VladWithCookies — APPROVED** [review]

**Pull request merged**